lib/src/radio-group/radio-group.component.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { ViewChild, Component, DebugElement } from "@angular/core";
+import { async, ComponentFixture, TestBed } from "@angular/core/testing";
+import { By } from "@angular/platform-browser";
+import { RadioGroupComponent, RadioGroupItem } from "./radio-group.component";
+import { FormsModule } from "@angular/forms";
+import { CommonModule } from "@angular/common";
+
+@Component({
+    selector: "test-sebng-radio-group",
+    template: `
+        <sebng-radio-group
+            id="myRadioGroup"
+            [list]="list"
+            [(ngModel)]="selectedItem"
+            [label]="label"
+            [className]="className"
+            [disabled]="disabled"
+            [condensed]="condensed"
+            [inline]="inline"
+        ></sebng-radio-group>
+    `,
+})
+class RadioGroupTestComponent {
+    @ViewChild(RadioGroupComponent) radioGroupComponent: RadioGroupComponent;
+    list: Array<RadioGroupItem>;
+    selectedItem: RadioGroupItem;
+    label?: string;
+    className?: string;
+    disabled?: boolean = false;
+    condensed?: boolean = false;
+    inline?: boolean = false;
+
+    constructor() {
+        this.list = [
+            { label: "Nigeria", value: "NG", key: "1" },
+            { label: "Malaysia", value: "MY", key: "2" },
+        ];
+    }
+}
+
+describe("RadioGroupComponent", () => {
+    let component: RadioGroupTestComponent;
+    let fixture: ComponentFixture<RadioGroupTestComponent>;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [FormsModule, CommonModule],
+            declarations: [RadioGroupComponent, RadioGroupTestComponent],
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(RadioGroupTestComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it("should create", () => {
+        expect(component).toBeTruthy();
+    });
+
+    it("Should render label", async(() => {
+        component.label = "Country";
+        fixture.detectChanges();
+        const label = fixture.debugElement.query(By.css(".radio-group-label > b"));
+        expect(label).not.toBeNull();
+        expect(label.nativeElement.innerHTML).toEqual("Country");
```

Comment:
I don't think the query returns `null` if it's not found, it will return `undefined` instead
```suggestion
        const label: HTMLLabelElement = fixture.debugElement.query(By.css(".radio-group-label"));
        expect(label).toBeDefined();
        expect(label.nativeElement.innerHTML).toEqual("<b>Country</b>");
```

Reply:
you mean `const label: DebugElement`

Reply:
Error: Expected '<b _ngcontent-a-c40=""'>Country</b>` to equal `<b>Country</b>`

Reply:
@kherP any suggestions?

Reply:
supposedly if not found, it will return null

Reply:
and if it is null, `expect(null).toBeDefined()` will pass the test, i believe in this `not.toBeNull()` is better

Reply:
@mario-subo 
Just use a regex `innerHTML.replace(/\<b ?(.*?)\>/g, "<b>")` I already updated the suggestion

Reply:
but why not just use `"label > b"` and check the string itself. I don't understand the unnecessary complexity. What does omitting `> b` and adding regex actually solve?

Reply:
Now I think about it, why do you have a `b` tag? if you're following bootstrap styles, a `label` tag doesn't need a `b` tag inside. If they chose to make it bold they will make it bold

See this example, we don't render `b` tag inside anymore
https://sebgroup.github.io/react-components/#/radiogroup

Reply:
we can omit the ` > b ` by setting the expect to `expect(label.nativeElement.innerHTML).toContain("Country");`

I don't understand why we need the regex as well haha

Reply:
yeah I will use your suggestion KP. Yousif as far as b inside the label goes I have been following that because the design used to be like that. I am thinking of removing the label altogether from all the components. I don't think it's needed. but as it stands now all my labels have a b in them so for now I will leave it and we can refactor all labels later.

Reply:
@KP suggestion is good, FYI you are already checking if b tag is a direct child of .radio-group-label here
const label = fixture.debugElement.query(By.css(".radio-group-label > b"));